aclk/agent_cloud_link.c

```diff
@@ -129,6 +161,31 @@ struct aclk_query_queue {
     u_int64_t count;
 } aclk_queue = { .aclk_query_head = NULL, .aclk_query_tail = NULL, .count = 0 };
 
+/*
+ * After a connection failure -- delay in milliseconds
+ * When a connection is established, the delay function
+ * should be called with mode 0 to reset the fail count
+ */
+unsigned long int aclk_delay(int mode)
+{
+    static int fail = 0;
+
+    if (!mode) {
+        fail = 0;
+        return 0;
+    }
+
+    if (fail == 0) {
+        srandom(time(NULL));
+        fail++;
+        return 0;
+    }
+
+    unsigned long int delay = (unsigned long int)((powl(ACLK_DELAY_SEED, fail)) * 1000 + (random() % 1618));
+
+    return MIN(delay, ACLK_MAX_BACKOFF_DELAY);
```

Comment:
It would be better to make the failure count saturate so that it sticks at the highest value in the sequence rather than cropping to another value.

Reply:
Will refine